libraries/WiFi/src/WiFiClient.cpp

```diff
@@ -304,7 +320,7 @@ int WiFiClient::connect(const char *host, uint16_t port)
 int WiFiClient::connect(const char *host, uint16_t port, int32_t timeout_ms)
 {
     IPAddress srv((uint32_t)0);
-    if(!WiFiGenericClass::hostByName(host, srv)){
+    if(!WiFiGenericClass::hostByName(host, srv, (WiFiGenericClass::getStatusBits() & WIFI_WANT_IP6_BIT))){
```

Comment:
Rather than change the signature, do the check for (WiFiGenericClass::getStatusBits() & WIFI_WANT_IP6_BIT) inside WiFiGeneric

Reply:
yes. left there to remind me to ask for discussion